mypyc/genops.py

```diff
@@ -2083,6 +2083,50 @@ def translate_refexpr_call(self, expr: CallExpr, callee: RefExpr) -> Value:
 
         # TODO: Generalize special cases
 
+        # Special case builtins.any
+        if (callee.fullname == 'builtins.any'
+                and len(expr.args) == 1
+                and expr.arg_kinds == [ARG_POS]
+                and isinstance(expr.args[0], GeneratorExpr)):
+            gen = expr.args[0]
+            retval = self.primitive_op(false_op, [], -1)
+            loop_params = list(zip(gen.indices, gen.sequences, gen.condlists))
+
+            def gen_inner_stmts() -> None:
+                true_block, false_block = BasicBlock(), BasicBlock()
+                comparison = self.accept(gen.left_expr)
+                self.add_bool_branch(comparison, true_block, false_block)
+                self.activate_block(true_block)
+                retval = self.primitive_op(true_op, [], -1)
+                self.add(Return(retval))
+                self.activate_block(false_block)
+
+            self.comprehension_helper(loop_params, gen_inner_stmts, expr.line)
+            return retval
+
+        # Special case builtins.all
+        if (callee.fullname == 'builtins.all'
+                and len(expr.args) == 1
+                and expr.arg_kinds == [ARG_POS]
+                and isinstance(expr.args[0], GeneratorExpr)):
+            gen = expr.args[0]
+            retval = self.primitive_op(true_op, [], -1)
+            loop_params = list(zip(gen.indices, gen.sequences, gen.condlists))
+
+            def gen_inner_stmts() -> None:
+                true_block, false_block = BasicBlock(), BasicBlock()
+                comparison = self.accept(gen.left_expr)
+                not_comparison = self.unary_op(comparison, 'not', expr.line)
+
+                self.add_bool_branch(not_comparison, true_block, false_block)
+                self.activate_block(true_block)
+                retval = self.primitive_op(false_op, [], -1)
+                self.add(Return(retval))
+                self.activate_block(false_block)
+
+            self.comprehension_helper(loop_params, gen_inner_stmts, expr.line)
```

Comment:
This looks solid. Two things:
1. Pull them out into functions
2. any and all are closely related, in a similar way to and and or. You might be able to merge their implementations in a similar way. Don't worry too much if there isn't something obvious though.